### DISCUSSION TITLE
Support for multiple accounts?

### DISCUSSION THREAD

**Discussion by Fayrewood** [BottlecapDave/HomeAssistant-OctopusEnergy#615] [Q&A]
I have multiple Octopus Accounts (one for each property and MPAN).  My API Key says it is applicable to all accounts.  When I try to add a second entry to the Integration, it asks for a target rate for my existing MPAN account.

Can I add a second account using the same API key with the Integration and how do I do that?

Many thanks!

**Comment by BottlecapDave** [comment]
Hello. I'm afraid the integration currently doesn't support multiple accounts. I've been chipping away to hopefully make this easier in the future to add, but the feature itself is low down on my personal list. Please upvote https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/223 so I can guage interest.

**Comment by RaoulSargent** [comment]
I currently have 4x (four) unique octopus accounts that I would like to track from the same home assistant instance. 

therefore need the ability to specify multiple individual instances of the octopus plugin (which is excellent btw), each with separate MPAN, octopus account id and api keys. 

Thank you.

**Comment by Dukeicon** [comment]
> I currently have 4x (four) unique octopus accounts that I would like to track from the same home assistant instance.
> 
> therefore need the ability to specify multiple individual instances of the octopus plugin (which is excellent btw), each with separate MPAN, octopus account id and api keys.
> 
> Thank you.

There is a BETA version which supports mutliple accounts.

How to install it via HACS is here: https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/223#issuecomment-1902083426

Note, that refers to BETA 1, BETA 2 has been release since - URL for that is: https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/multi-account-support-beta-2
